tests/e2e_test.go

```diff
@@ -80,8 +80,6 @@ func TestProviderE2E(t *testing.T) {
 			group := "testgroup"
 			app := "testapp"
 
-			promoteBranchName := fmt.Sprintf("promote/%s-%s", group, app)
```

Comment:
Don't we need the promote branch name for dev?

Reply:
It seems the e2e test doesn't verify that the `new` command creates the branch? Perhaps it should, though?

Reply:
I was confused, of course. Reworked this a bit, so now we check each step properly.